### PR TITLE
[ECO-1700] Add `app/emojicoin/[market]page.tsx` slug and various frontend refactors

### PR DESCRIPTION
**Pull request by xbtmatt**
# Description

 - [x] Add server-side fetching for:
   - [x] Candlesticks
   - [x] Markets
   - [x] Market data
   - [x] Chat data
   - [x] Anything else relevant
     - [x] `get_top_markets`, for the `Home` page (`/` and `/home`) (see sql query below)
     - [x] `fetchLastMarketState`, gets the `State` and `MarketView` for a market slug
 - [x] Reroute emojicoin to emojicoin/0
 - [x] Add `not-found.tsx` page to `emojicoin`
 - [x] Use [market] in slug for query params
   - [x] ~Resolve to a symbol or a marketID~ should be for another PR

TODO: Follow-up PR for `inbox` to add this query:

```sql
CREATE OR REPLACE FUNCTION get_top_markets(
    module_address text
)
RETURNS TABLE (
    transaction_version bigint,
    data jsonb
) AS $$
BEGIN
    RETURN QUERY
    WITH latest_entries AS (
        SELECT DISTINCT ON (e.data->'market_metadata'->'market_id') e.*
        FROM public.inbox_events e
        WHERE e.type = module_address || '::emojicoin_dot_fun::State'
        ORDER BY e.data->'market_metadata'->'market_id', (e.data->'last_swap'->>'time')::bigint DESC
    )
    SELECT
        le.transaction_version,
        le.data
    FROM latest_entries le
    ORDER BY (le.data->'instantaneous_stats'->>'market_cap')::bigint DESC;
END;
$$ LANGUAGE plpgsql;
```

# Testing

TBD

# Checklist

- [x] Did you update relevant documentation?
- [x] Did you add tests to cover new code or a fixed issue?
- [x] Did you update the changelog?
- [x] Did you check all checkboxes from the linked Linear task?

### CODE REVIEW

File: src/typescript/sdk/src/utils/mock-data.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ async function main() {
             user: t,
             tradeInputs: {
               inputAmount: amt,
-              isSell: amt < 0n,
+              // isSell: amt < 0n, // Determined automatically by `amt` < 0n.
```
Comment (xbtmatt) [inline src/typescript/sdk/src/utils/mock-data.ts:131]:
```suggestion
```

File: src/typescript/sdk/src/queries/swap.ts [hunk 2]
```diff
@@ -24,18 +24,20 @@ export const paginateUserSwapEvents = async (
...
-  return {
-    events: res.data.map((e) => toSwapEvent(e)),
-    errors: res.errors,
-  };
+  if (errors.some((e) => e)) {
+    /* eslint-disable-next-line no-console */
+    console.warn("Error fetching chat events", errors);
+  }
+
+  return data;
```
Comment (xbtmatt) [inline src/typescript/sdk/src/queries/swap.ts:40]: Consider adding this to some of the other fetches, eventually..?

File: src/typescript/sdk/src/queries/market.ts [hunk 3]
```diff
@@ -1,12 +1,38 @@
...
+      module_address: MODULE_ADDRESS.toString(),
+    })
+    .select("*");
+  const { data } = res;
+  return {
+    data: data ? data.map((v) => ({ data: v.data, version: v.transaction_version })) : [],
+    error: res.error,
+  };
+};
+
```
Comment (xbtmatt) [inline src/typescript/sdk/src/queries/market.ts:35]: Need to get the sql extension for this query into an `inbox` PR

File: src/typescript/frontend/next.config.mjs [hunk 4]
```diff
@@ -1,36 +1,54 @@
...
+  redirects: async () => {
+    return [
+      {
+        source: "/emojicoin",
+        destination: "/emojicoin/0",
+        permanent: true,
+      },
+    ];
+  },
+};
```
Comment (xbtmatt) [inline src/typescript/frontend/next.config.mjs:52]: Remove debugging options before merging.

File: src/typescript/sdk/src/utils/mock-data.ts [hunk 5]
```diff
@@ -128,7 +127,7 @@ async function main() {
...
```
Comment (xbtmatt) [inline src/typescript/sdk/src/utils/mock-data.ts:130]:
```suggestion
```

**Review by alnoki — APPROVED** [review]

**Pull request merged**